app/controllers/admin/student_accounts_controller.rb

```diff
@@ -0,0 +1,37 @@
+require 'date'
+
+module Admin
+    class StudentAccountsController < BaseController
+      layout 'admin'
+  
+      def actions
+      end
+  
+      def get_number_of_accounts
+        days_num = get_number_of_days
+        accounts_num = query(days_num)
+        flash[:notice] = "The number of student acccounts created since July 1 is " + accounts_num.to_s
+        redirect_to actions_admin_student_accounts_path
+      end
+
+      def get_number_of_days
+        # number of days since July 1
+        today = Date.new
+        year = today.year
+        month = today.month
+        if month < 6
+            year = year - 1
+        end
+        #July is a keyword, so using month7
+        month7 = Date.parse year.inspect + "-07-01"
+        return (month7...today).count
+      end
```

Comment:
See my other comment below where I suggest removing this function.

Reply:
I still need to calculate which year to use in the query. I renamed the function and removed unneeded code.

Reply:
@edwoodward I looked at this and thought "every time I try some custom code like this I find that Rails has already done it somewhere", so I looked around, but I didn't find it! So congrats on finding a hole! :-)

We'll want to test this code somewhere, and testing utility methods in a controller is kind of a heavy lift, so I think a better thing to do might be to monkey patch (extend) the `DateTime` class.  

E.g. you could make a file in `lib/date_time.rb` that has:

```ruby
class DateTime        # in Ruby you can reopen a class and add more stuff to it
  def self.last_july_1
    # pasting code here from your `get_year` and `query` methods
    today = DateTime.now
    year = today.year
    month = today.month
    if month < 7
      year = year - 1
    end
    # Should probably use `DateTime` here since rails dates are DateTime objects, and 
    # can also use string interpolation, i.e. `DateTime.parse("#{year}-07-01")`
    Date.parse(year.inspect + '-07-01')    
  end
end
```

Instead of `last_july_1` a name that fits other [`DateTime` methods](https://api.rubyonrails.org/classes/DateTime.html) might be `beginning_of_rice_fiscal_year` or `beginning_of_last_q3` or `beginning_of_last_july`.

Files in the `lib` directory are not automatically read by Rails like those in `app` are.  To make sure the code above is seen by Rails, you'd `require` it from some other file that Rails already requires.  What we typically use is `config/environment.rb` -- if you go here you'll see [a bunch of `require` statements](https://github.com/openstax/accounts/blob/master/config/environment.rb#L4-L22) for files in the `lib` directory.  

With this `DateTime.last_july_1` method in hand, you can write a spec in `spec/lib/date_time_spec.rb` (a file you'd make) to test that it gives the right datetime.  To test it, you'll want to make the method see fake values for `DateTime.now` in your code (so that you can test that "last July 1" is different if "now" is Sept 28 vs April 1.  In rspec, the way to do this is to mock the return value for `DateTime.now` with a line such as:

```ruby
allow(DateTime).to receive(:now).and_return(DateTime.parse("2019-09-28"))
```

You'll see examples of this all over the specs.  You can run this individual spec file with:

```
$> bundle exec rspec ./spec/lib/date_time_spec.rb
```

With the spec working, see below for a comment that talks about using your new method.

Reply:
DateTime function created and spec added